packit_service/worker/handlers/testing_farm.py

```diff
@@ -310,7 +310,14 @@ def run(self) -> TaskResults:
             logger.warning(msg)
             return TaskResults(success=False, details={"msg": msg})
 
-        test_run_model.set_status(self.result, created=self.created)
+        if test_run_model.status == self.result:
+            logger.debug(
+                "Testing farm results already processed "
+                "(state in the DB is the same as the one about to report)."
+            )
+            return TaskResults(
+                success=True, details={"msg": "Testing farm results already processed"}
+            )
```

Comment:
Can't we lose commit statuses with this approach?

(E.g. Handler will update the database and is killed right after that.)

I would either don't skip too much to not lose anything or do the same thing twice (can't hurt the UX).

Reply:
I moved the setting of the status in DB after reporting the status to the user to prevent that, WDYT?

Reply:
:+1: